src/conversions.jl

```diff
@@ -5,11 +5,76 @@ using Mocking: Mocking, @mock
 const utc_tz = FixedTimeZone("UTC")
 
 """
-    DateTime(::ZonedDateTime) -> DateTime
+    DateTime(zdt::ZonedDateTime, ::Union{Type{Local}, Type{UTC}}) -> DateTime
 
-Returns an equivalent `DateTime` without any `TimeZone` information.
+Create an implicit local/UTC `DateTime` from the given `ZonedDateTime`.
+
+# Example
+
+```jldoctest
+julia> zdt = ZonedDateTime(2014, 5, 30, 21, tz"UTC-4")
+2014-05-30T21:00:00-04:00
+
+julia> DateTime(zdt, Local)
+2014-05-30T21:00:00
+
+julia> DateTime(zdt, UTC)
+2014-05-31T01:00:00
+```
 """
-DateTime(zdt::ZonedDateTime) = localtime(zdt)
+DateTime(::ZonedDateTime, ::Union{Type{Local}, Type{UTC}})
+
+Dates.DateTime(zdt::ZonedDateTime, ::Type{Local}) = zdt.utc_datetime + zdt.zone.offset
+Dates.DateTime(zdt::ZonedDateTime, ::Type{UTC}) = zdt.utc_datetime
+
+
+"""
+    Date(zdt::ZonedDateTime, ::Union{Type{Local}, Type{UTC}}) -> Date
+
+Create an implicit local/UTC `Date` from the given `ZonedDateTime`.
+
+# Example
+
+```jldoctest
+julia> zdt = ZonedDateTime(2014, 5, 30, 21, tz"UTC-4")
+2014-05-30T21:00:00-04:00
+
+julia> Date(zdt, Local)
+2014-05-30
+
+julia> Date(zdt, UTC)
+2014-05-31
+```
+"""
+Date(::ZonedDateTime, ::Union{Type{Local}, Type{UTC}})
+
+Dates.Date(zdt::ZonedDateTime, ::Type{Local}) = Date(DateTime(zdt, Local))
+Dates.Date(zdt::ZonedDateTime, ::Type{UTC}) = Date(DateTime(zdt, UTC))
+
+
+"""
+    Time(zdt::ZonedDateTime, ::Union{Type{Local}, Type{UTC}}) -> Time
+
+Create an implicit local/UTC `Time` from the given `ZonedDateTime`.
```

Comment:
Couild we add a bit more text?

```suggestion
Create an implicit local/UTC `Time` from the given `ZonedDateTime`.

 - `Local` means a time local to the timezone in the `ZonedDateTime`, i.e. just discarding the timezone information. (**Not** local to the system time of the user invoking the code).
  - `UTC` means to shift to the `UTC` timezone and then discard all timezone information.
```

Maybe also this particular text should be put in to a string constant and interpolatred in at a few places?
Like in `Date` and `DateTime` as well?

Probably could be written better

Reply:
I rewrote the docstrings to be more sensible

Reply:
I rewrote the docstrings to be more sensible